cli/command/service/create.go

```diff
@@ -39,6 +39,8 @@ func newCreateCommand(dockerCli *command.DockerCli) *cobra.Command {
 	flags.StringSliceVar(&opts.constraints, flagConstraint, []string{}, "Placement constraints")
 	flags.StringSliceVar(&opts.networks, flagNetwork, []string{}, "Network attachments")
 	flags.VarP(&opts.endpoint.ports, flagPublish, "p", "Publish a port as a node port")
```

Comment:
Given that we have 2 independent publish flags, should we move -p above to hidden ?
`flags.MarkHidden("p")`